### PR TITLE
Meet Shah: Lesson02

### PR DESCRIPTION
**Pull request by meet29in**

### CODE REVIEW

File: students/meet_shah/lesson02/print_grid.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
+def print_grid(x):
```
Comment (HarryMaher) [inline students/meet_shah/lesson02/print_grid.py]: Looks great!

File: students/meet_shah/lesson02/fizzbuzz.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+def fizzbuzz():
```
Comment (HarryMaher) [inline students/meet_shah/lesson02/fizzbuzz.py]: Nice! Good idea.

File: students/meet_shah/lesson02/print_grid.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    topstart = "+ "
+    topmid   = "- "
+    topend   = "+"
+
+    midstart = "| "
+    midmid   = "  "
+    midend   = "|"
+
+    repeat = int(x/2)
+    topline = topstart + topmid*repeat + topstart + topmid*repeat + topend
```
Comment (HarryMaher) [inline students/meet_shah/lesson02/print_grid.py]: You can multiply help make it more DRY and less typing. Eg:
```
topline = (topstart + topmid * repeat) * 2 + topend
```

File: students/meet_shah/lesson02/print_grid.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        for i in range(0,repeat):
+            print(midline)
+        print(topline)
+ 
+def help_print_line(startpattern, endpattern, repeat):
+    for i in range(0,repeat):
+        print(startpattern, end='')
+    print(endpattern)
+
+def print_grid2(box_repeat,grid_repeat):
```
Comment (HarryMaher) [inline students/meet_shah/lesson02/print_grid.py]: Awesome!

File: students/meet_shah/lesson02/print_grid.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    midline = midstart + midmid*repeat + midstart + midmid*repeat + midend
+    
+    print(topline)
+    for i in range(0,2):
+        for i in range(0,repeat):
+            print(midline)
+        print(topline)
+ 
+def help_print_line(startpattern, endpattern, repeat):
+    for i in range(0,repeat):
```
Comment (HarryMaher) [inline students/meet_shah/lesson02/print_grid.py]: I like that you used a helper function to break up the code!
You can simplify this a bit by getting rid of the loop here and just using string multiplication. 
``` 
ret = startpattern * repeat + endpattern
print(ret)
```

File: students/meet_shah/lesson02/print_grid.py (hunk 3 above)
Comment (meet29in) [reply students/meet_shah/lesson02/print_grid.py]: ah sweet! totally missed that.

**Comment by meet29in** [comment]
I pushed part3 late Sunday night. It's not past due yet but I was curious if it looked good or if I need to the code here.

File: students/meet_shah/lesson02/series.py [hunk 6]
```diff
@@ -0,0 +1,32 @@
+
+def fib(n):
```
Comment (HarryMaher) [inline students/meet_shah/lesson02/series.py]: Cool, looks good. We also wanted a sum_series function, but this look like you've good the idea. Check out the solution we did in class if that was causing you issues.

Comment (meet29in) [reply students/meet_shah/lesson02/series.py]: yep missed the sum_series :(

**Pull request merged**